var/spack/repos/builtin/packages/cosmomc/package.py

```diff
@@ -0,0 +1,192 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import fnmatch
+import os
+import shutil
+
+
+class Cosmomc(Package):
+    """CosmoMC is a Fortran 2008 Markov-Chain Monte-Carlo (MCMC) engine
+       for exploring cosmological parameter space, together with
+       Fortran and python code for analysing Monte-Carlo samples and
+       importance sampling (plus a suite of scripts for building grids
+       of runs, plotting and presenting results)."""
+
+    homepage = "http://cosmologist.info/cosmomc/"
+    url      = "https://github.com/cmbant/CosmoMC/archive/Nov2016.tar.gz"
+
+    version('Nov2016',  '98620cb746352f68fb0c1196e9a070ac', preferred=True)
```

Comment:
it's better to rename to `2016.11` so that versions have correct order. You would need a custom `url_for_version`, see for a similar example https://github.com/LLNL/spack/blob/develop/var/spack/repos/builtin/packages/lammps/package.py#L39-L42